src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -152,9 +152,9 @@ internal class DependencyResolver
             }
 
             // Link to dependent repo, don't build the file, leave href as is
-            if (file.FilePath.Origin == FileOrigin.Dependency)
+            if (file.FilePath.Origin == FileOrigin.Dependency && !_buildScope.InScopeDependencyNames.Contains(file.FilePath.DependencyName))
```

Comment:
Checking if a dependency is `in scope` can be done by DependencyResolver itself without asking BuildScope.

Reply:
Don't want to make `BuildScope` and `DependencyResolver` both understand `in scope`, logically it belongs to `BuildScope`